packages/core/contracts/cross-chain-oracle/chain-adapters/Optimism_ChildMessenger.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+// This should be replaced with a "real" import when Optimism release their new contract versions.
+import "../../external/ovm/OVM_CrossDomainEnabled.sol";
+import "../interfaces/ChildMessengerInterface.sol";
+import "../interfaces/ChildMessengerConsumerInterface.sol";
+
+/**
+ * @notice Sends cross chain messages from Optimism L2 to Ethereum L1 network.
+ * @dev This contract's is ownable via the onlyCrossDomainAccount modifier, restricting ownership to the cross-domain
+ * parent messenger contract that lives on L1.
+ */
+contract Optimism_ChildMessenger is OVM_CrossDomainEnabled, ChildMessengerInterface {
+    // The only L2 contract that can send messages over the bridge via the messenger is the oracle spoke.
+    address public oracleSpoke;
+
+    // Messenger contract on the other side of the L1<->L2 bridge.
+    address public parentMessenger;
+
+    // Hard coded default gas limit for L1 transactions.
+    uint32 public defaultGasLimit = 5_000_000;
+
+    // TODO: import from optimism contracts when they release their latest version.
+    address internal constant L2_CROSS_DOMAIN_MESSENGER = 0x4200000000000000000000000000000000000007;
+
+    event OracleSpokeSet(address newOracleSpoke);
```

Comment:
nit: you're going to hate me, but usually we've been naming events as VerbNoun. For example these events should probably be renamed:

```
SetOracleSpoke
SetParentMessenger
SetDefaultGasLimit
```

On the MessageSent and MessagReceived events I'm good with leaving those as-is because they both conveniently start with "Message"

Reply:
ok.